spring-cloud-huawei-config/src/main/java/com/huaweicloud/config/ConfigWatch.java

```diff
@@ -48,17 +48,18 @@ public void setApplicationEventPublisher(ApplicationEventPublisher applicationEv
 
   @Subscribe
   public void onConfigurationChangedEvent(ConfigurationChangedEvent event) {
-    LOGGER.info("receive new configurations, added=[{}], updated=[{}], deleted=[{}]",
-        event.getAdded().keySet(),
-        event.getUpdated().keySet(),
-        event.getDeleted().keySet());
-
+    Set<String> addKeys = event.getAdded().keySet();
+    Set<String> updateKeys = event.getUpdated().keySet();
+    Set<String> deleteKeys = event.getDeleted().keySet();
+    if (addKeys.isEmpty() && updateKeys.isEmpty() && deleteKeys.isEmpty()) {
+      return;
+    }
+    LOGGER.info("receive new configurations, added=[{}], updated=[{}], deleted=[{}]", addKeys, updateKeys, deleteKeys);
     Set<String> updatedKey = new HashSet<>();
     updatedKey.addAll(event.getAdded().keySet());
     updatedKey.addAll(event.getUpdated().keySet());
     updatedKey.addAll(event.getDeleted().keySet());
     ConfigRefreshEvent configRefreshEvent = new ConfigRefreshEvent(this, updatedKey);
-    applicationEventPublisher.publishEvent(new ConfigRefreshEvent(this, updatedKey));
```

Comment:
Can not remove this statement. And ConfigRefreshEvent should send after RefreshEvent